config.go

```diff
@@ -1,31 +1,133 @@
 package main
 
-import "github.com/getAlby/nostr-wallet-connect/models/db"
+import (
+	"crypto/rand"
+	"encoding/hex"
+	"os"
+	"time"
+
+	"gorm.io/gorm"
+	"gorm.io/gorm/clause"
+)
 
 const (
 	LNDBackendType   = "LND"
 	BreezBackendType = "BREEZ"
 	CookieName       = "alby_nwc_session"
 )
 
+type AppConfig struct {
+	Relay           string `envconfig:"RELAY" default:"wss://relay.getalby.com/v1"`
+	LNBackendType   string `envconfig:"LN_BACKEND_TYPE"`
+	LNDCertFile     string `envconfig:"LND_CERT_FILE"`
+	LNDMacaroonFile string `envconfig:"LND_MACAROON_FILE"`
+	Workdir         string `envconfig:"WORK_DIR" default:".data"`
+	Port            string `envconfig:"PORT" default:"8080"`
+	DatabaseUri     string `envconfig:"DATABASE_URI" default:"nostr-wallet-connect.db"`
+	CookieSecret    string `envconfig:"COOKIE_SECRET"`
+}
+
+type UserConfig struct {
+	ID        uint   `gorm:"primaryKey"`
+	Key       string `gorm:"unique;uniqueIndex;not null"`
+	Value     string
+	Encrypted bool
+	CreatedAt time.Time
+	UpdatedAt time.Time
+}
+
 type Config struct {
-	// These config can be set either by .env or the database config table.
-	// database config always takes preference.
-	db.Config
-	CookieSecret            string `envconfig:"COOKIE_SECRET"`
-	CookieDomain            string `envconfig:"COOKIE_DOMAIN"`
-	ClientPubkey            string `envconfig:"CLIENT_NOSTR_PUBKEY"`
-	Relay                   string `envconfig:"RELAY" default:"wss://relay.getalby.com/v1"`
-	PublicRelay             string `envconfig:"PUBLIC_RELAY"`
-	LNDCertFile             string `envconfig:"LND_CERT_FILE"`
-	LNDMacaroonFile         string `envconfig:"LND_MACAROON_FILE"`
-	BreezWorkdir            string `envconfig:"BREEZ_WORK_DIR" default:".breez"`
-	BasicAuthUser           string `envconfig:"BASIC_AUTH_USER"`
-	BasicAuthPassword       string `envconfig:"BASIC_AUTH_PASSWORD"`
-	Port                    string `envconfig:"PORT" default:"8080"`
-	DatabaseUri             string `envconfig:"DATABASE_URI" default:"nostr-wallet-connect.db"`
-	DatabaseMaxConns        int    `envconfig:"DATABASE_MAX_CONNS" default:"10"`
-	DatabaseMaxIdleConns    int    `envconfig:"DATABASE_MAX_IDLE_CONNS" default:"5"`
-	DatabaseConnMaxLifetime int    `envconfig:"DATABASE_CONN_MAX_LIFETIME" default:"1800"` // 30 minutes
-	IdentityPubkey          string
+	Env            *AppConfig
+	CookieSecret   string
+	NostrSecretKey string
+	NostrPublicKey string
+	db             *gorm.DB
+}
+
+func (cfg *Config) Init(db *gorm.DB, env *AppConfig) {
+	cfg.db = db
+	cfg.Env = env
+
+	if cfg.Env.Relay != "" {
+		cfg.SetUpdate("Relay", cfg.Env.Relay, "")
```

Comment:
Why do we even set it in the DB and not just use cfg.Env.Relay?

Reply:
no idea. was wondering that it could be configurable if I as a user want to use a different one.